RealmSwift/RealmKeyedCollection.swift

```diff
@@ -0,0 +1,856 @@
+
+     - parameter value: The object value.
+     - parameter key:   The name of the property whose value should be set on each object.
+    */
+    func setValue(_ value: Any?, forKey key: String)
+
+    // MARK: Filtering
+
+    /**
+     Returns a `Results` containing all matching key-value pairs the given predicate in the Map.
+
+     - parameter predicate: The predicate with which to filter the objects.
+     */
+    func filter(_ predicate: NSPredicate) -> Results<Value>
+
+    /**
+     Returns a Boolean value indicating whether the Map contains the key-value pair
+     satisfies the given predicate
+
+     - parameter where: a closure that test if any key-pair of the given map represents the match.
+     */
+    func contains(where predicate: @escaping (_ key: Key, _ value: Value) -> Bool) -> Bool
+
+    // MARK: Sorting
+
+    /**
+     Returns a `Results` containing the objects in the dictionary, but sorted.
+
+     Objects are sorted based on their values. For example, to sort a dictionary of `Date`s from
+     neweset to oldest based, you might call `dates.sorted(ascending: true)`.
+
+     - parameter ascending: The direction to sort in.
+     */
+    func sorted(ascending: Bool) -> Results<Value>
+
+    /**
+     Returns a `Results` containing the objects in the dictionary, but sorted.
+
+     Objects are sorted based on the values of the given key path. For example, to sort a dictionary of `Student`s from
+     youngest to oldest based on their `age` property, you might call
+     `students.sorted(byKeyPath: "age", ascending: true)`.
+
+     - warning: Dictionaries may only be sorted by properties of boolean, `Date`, `NSDate`, single and double-precision
+                floating point, integer, and string types.
+
+     - parameter keyPath:  The key path to sort by.
+     - parameter ascending: The direction to sort in.
+     */
+    func sorted(byKeyPath keyPath: String, ascending: Bool) -> Results<Value>
+
+    /**
+     Returns a `Results` containing the objects in the dictionary, but sorted.
+
+     - warning: Dictionaries may only be sorted by properties of boolean, `Date`, `NSDate`, single and double-precision
+                floating point, integer, and string types.
+
+     - see: `sorted(byKeyPath:ascending:)`
+    */
+    func sorted<S: Sequence>(by sortDescriptors: S) -> Results<Value>
+        where S.Iterator.Element == SortDescriptor
+
+    /// Returns all of the keys in this dictionary.
+    var keys: [Key] { get }
+
+    /// Returns all of the values in the dictionary.
+    var values: [Value] { get }
+
+    /**
+     Returns the object at the given `position`.
+
+     - parameter position: The position of the element in the collection.
+     */
+    subscript(position: MapIndex) -> (key: Key, value: Value) { get }
+
+    /// Returns the position of an element in the collection.
+    /// - Parameter object: The object to find.
+    func index(of object: Value) -> MapIndex?
```

Comment:
Do we actually need to expose these? If we do the docs need a bunch of caveats about how dictionary orders are unstable and not the insertion order.

Reply:
No, I added them because they are a feature of Foundation, but I was hard strung to find real world examples of where they are actually used. I'll remove.